From 5617a1303210d201e30b56e79c40273784833334 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Fri, 6 Dec 2013 17:20:19 +0800
Subject: [PATCH 269/310] ENGR00291127 Make EVK 6SL can be built out without
 EULA gits

Only include GPU/Sensor related makefile if it is exist
Move vpu firmware related copyfile to board's makefile

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 evk_6sl/AndroidBoard.mk | 4 ++--
 evk_6sl/BoardConfig.mk  | 2 +-
 imx6/arm2_6dq.mk        | 4 ++++
 imx6/hdmidongle_6dq.mk  | 4 ++++
 imx6/imx6.mk            | 5 -----
 imx6/sabreauto_6q.mk    | 4 ++++
 imx6/sabresd_6dq.mk     | 4 ++++
 7 files changed, 19 insertions(+), 8 deletions(-)

diff --git a/evk_6sl/AndroidBoard.mk b/evk_6sl/AndroidBoard.mk
index 3368151..0b14928 100644
--- a/evk_6sl/AndroidBoard.mk
+++ b/evk_6sl/AndroidBoard.mk
@@ -1,5 +1,5 @@
 LOCAL_PATH := $(call my-dir)
 ifeq ($(PREBUILT_FSL_IMX_CODEC),true)
-include device/fsl-proprietary/codec/fsl-codec.mk
+-include device/fsl-proprietary/codec/fsl-codec.mk
 endif
-include device/fsl-proprietary/sensor/fsl-sensor.mk
+-include device/fsl-proprietary/sensor/fsl-sensor.mk
diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index ebf4e7b..2716c22 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -5,7 +5,7 @@
 include device/fsl/imx6/soc/imx6sl.mk
 include device/fsl/evk_6sl/build_id.mk
 include device/fsl/imx6/BoardConfigCommon.mk
-include device/fsl-proprietary/gpu-viv/fsl-gpu.mk
+-include device/fsl-proprietary/gpu-viv/fsl-gpu.mk
 # evk_mx6sl default target for EXT4
 BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
diff --git a/imx6/arm2_6dq.mk b/imx6/arm2_6dq.mk
index e730994..2d3addb 100644
--- a/imx6/arm2_6dq.mk
+++ b/imx6/arm2_6dq.mk
@@ -19,6 +19,10 @@ PRODUCT_COPY_FILES += \
 	device/fsl/arm2_6dq/required_hardware.xml:system/etc/permissions/required_hardware.xml \
 	device/fsl/arm2_6dq/init.rc:root/init.freescale.rc
 
+PRODUCT_COPY_FILES +=	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6d.bin:system/lib/firmware/vpu/vpu_fw_imx6d.bin 	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6q.bin:system/lib/firmware/vpu/vpu_fw_imx6q.bin
+
 DEVICE_PACKAGE_OVERLAYS := device/fsl/arm2_6dq/overlay
 
 PRODUCT_CHARACTERISTICS := tablet
diff --git a/imx6/hdmidongle_6dq.mk b/imx6/hdmidongle_6dq.mk
index b76fe3f..139b9b6 100644
--- a/imx6/hdmidongle_6dq.mk
+++ b/imx6/hdmidongle_6dq.mk
@@ -22,6 +22,10 @@ PRODUCT_COPY_FILES += \
 	device/fsl/hdmidongle_6dq/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/hdmidongle_6dq/audio_effects.conf:system/vendor/etc/audio_effects.conf
 
+PRODUCT_COPY_FILES +=	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6d.bin:system/lib/firmware/vpu/vpu_fw_imx6d.bin 	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6q.bin:system/lib/firmware/vpu/vpu_fw_imx6q.bin
+
 # GPU files
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/hdmidongle_6dq/overlay
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 555bae9..670bcf1 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -329,11 +329,6 @@ PRODUCT_COPY_FILES +=	\
 	device/fsl/common/display/display_mode_fb2.conf:system/etc/display_mode_fb2.conf \
 	device/fsl/common/display/display_mode_fb4.conf:system/etc/display_mode_fb4.conf
 
-
-PRODUCT_COPY_FILES +=	\
-	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6d.bin:system/lib/firmware/vpu/vpu_fw_imx6d.bin 	\
-	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6q.bin:system/lib/firmware/vpu/vpu_fw_imx6q.bin
-
 # we have enough storage space to hold precise GC data
 PRODUCT_TAGS += dalvik.gc.type-precise
 
diff --git a/imx6/sabreauto_6q.mk b/imx6/sabreauto_6q.mk
index 9474f81..6d59582 100644
--- a/imx6/sabreauto_6q.mk
+++ b/imx6/sabreauto_6q.mk
@@ -22,6 +22,10 @@ PRODUCT_COPY_FILES += \
 	device/fsl/sabreauto_6q/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/sabreauto_6q/audio_effects.conf:system/vendor/etc/audio_effects.conf
 
+PRODUCT_COPY_FILES +=	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6d.bin:system/lib/firmware/vpu/vpu_fw_imx6d.bin 	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6q.bin:system/lib/firmware/vpu/vpu_fw_imx6q.bin
+
 # GPU files
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/sabreauto_6q/overlay
diff --git a/imx6/sabresd_6dq.mk b/imx6/sabresd_6dq.mk
index c43cad0..8d760f8 100644
--- a/imx6/sabresd_6dq.mk
+++ b/imx6/sabresd_6dq.mk
@@ -23,6 +23,10 @@ PRODUCT_COPY_FILES += \
 	device/fsl/sabresd_6dq/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/sabresd_6dq/audio_effects.conf:system/vendor/etc/audio_effects.conf
 
+PRODUCT_COPY_FILES +=	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6d.bin:system/lib/firmware/vpu/vpu_fw_imx6d.bin 	\
+	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6q.bin:system/lib/firmware/vpu/vpu_fw_imx6q.bin
+
 # GPU files
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/sabresd_6dq/overlay
-- 
1.8.0

